include/random_walks/gaussian_accelerated_billiard_walk.hpp

```diff
@@ -177,33 +192,60 @@ struct GaussianAcceleratedBilliardWalk
                 }
 
                 _lambda_prev = dl * pbpair.first;
-                _p += (_lambda_prev * _v);
+                if constexpr (std::is_same<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>>::value) {
+                    _update_parameters.moved_dist = _lambda_prev;
+                    NT* Ar_data = _lambdas.data();
+                    NT* Av_data = _Av.data();
+                    for(int i = 0; i < P.num_of_hyperplanes(); ++i) {
+                        _distances_set.vec[i].first = ( *(b_data + i) - (*(Ar_data + i)) ) / (*(Av_data + i));
+                    }
+                    // rebuild the heap with the new values of (b - Ar) / Av
+                    _distances_set.rebuild(_update_parameters.moved_dist);
```

Comment:
Why not inserting the new values in the heap (in O(logn)) instead of rebuilding (in O(n))?

Reply:
here, this happens after we set a new direction, in which case now all the values have changed, so it's quicker to rebuild (O(n)) rather than insert each one in (O(nlogn)). I'm not entirely sure if it makes a difference, but I think it does, since afterwards I never have O(nlogn) things, just O(non_zeroes * logn), so this O(nlogn) might be a bottleneck